Makefile

```diff
@@ -215,6 +215,14 @@ rcc:
 	./ui/generate-rcc.sh
 	rcc --binary ui/resources.qrc -o ./resources.rcc
 
+# default token is a free-tier token with limited capabilities and usage
+# limits; our docs should include directions for community contributor to setup
+# their own Infura account and token instead of relying on this default token
+# during development
+DEFAULT_TOKEN := df3bd604ac4b4073823e72f6617fc7db
+INFURA_TOKEN ?= $(DEFAULT_TOKEN)
```

Comment:
Your issue is that you're missing `export` in this line. Makefile variablse are NOT by default available to make targets in their env. You need to `export` them in order for them to be inherited by processes started by `make` targets.

Reply:
It’s not relying on export: 

NIM_PARAMS += -d:INFURA_TOKEN:"$(INFURA_TOKEN)"

Instead it becomes part of a `-d` flag and gets baked into the binary.

Reply:
Oh, interesting... Windows is different because we don't run `make nim_status_client` separately. Maybe that's why?

Reply:
It seems that was it. See the latest commit I pushed to this PR branch.

So what I think was happening: targets like `$(STATUS_CLIENT_APPIMAGE)` depend on the phony target `nim_status_client`, which always resulted in a rebuild of `bin/nim_status_client` but without the `INFURA_TOKEN`.

I blame myself, but I've learned a ton about `make` stuff since I worked on this repo's Makefile and a refactor is planned anyway because of requirements that will be introduced by the `nim-status` library. I'll clean up the Makefile before too long and in the meantime I I think cutting out the explicit step of building `nim_status_client` is good enough, if you're fine with it.

Now that this build aspect has been sorted, I need to introduce a couple more things in the Nim sources and figure out why the app isn't launching correctly on my Windows laptop.

Reply:
I think it's fine to drop the stage that builds just the client.

Truth be told a `Makefile` that is more than just target definitions is a horrible thing. Doing complex logic in `Makefile`s is a painful thing indeed. If I cared enough to argue with everyone about it I'd say this should be built with Nix. This way you can divide everything into its own derivation - including the Nim compiler itself - and seriously cut down on build times, since all that is already built will not be rebuilt and will be fetched from cache.